src/app/features/settings/sign-out/sign-out-confirm.tsx

```diff
@@ -0,0 +1,36 @@
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { RouteUrls } from '@shared/route-urls';
+
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { useKeyActions } from '@app/common/hooks/use-key-actions';
+
+import { SignOutDialog } from './sign-out';
+
+interface SignOutProps {
+  isShowing: boolean;
+  onClose(): void;
+}
+
+export function SignOut({ isShowing = false, onClose }: SignOutProps) {
+  const analytics = useAnalytics();
+
+  //  TODO 4370 check if this is enough to track views of previous sign-out route
+  useEffect(() => void analytics.track('sign-out'), [analytics]);
+  const { signOut } = useKeyActions();
+  const navigate = useNavigate();
+  // FIXME same bug as SwitchAcccount dialog where we call hooks from useWalletType when no wallet yet set
```

Comment:
Do you need help testing these?

Reply:
I can re-test this and try to fix if I can, thanks. 

This is the bug I raised on the dev call where the wallet crashes on fresh install if you don't do 
```
if (!isShowing) return null
```

In some of the dialogs. The page crashes and you get 

```
Wallet is neither type software or ... 
```

I could be just overthinking it for this PR as with that early return it doesn't crash. The note more relates to why is this code even running at that point in the app lifecycle as those components shouldn't load until needed